content/ui-patterns/forms.mdx

```diff
@@ -0,0 +1,393 @@
+---
+title: Forms
+---
+
+import {Box, Heading, Link, Text} from '@primer/components'
+
+<Box sx={{fontSize: 3}} class="lead" as="p">
+    Forms are used to complete tasks that require data input from the user. For example: creating a new repo, configuring settings, and logging in.
+</Box>
+
+<Box sx={{fontSize: 3}} class="lead" as="p">
+    Primer's form design guidelines aim to minimize the effort and cognitive load required to complete a task that involves a form.
+</Box>
+
+## Form control anatomy
+
+![diagrams labeling the anatomy of a text field and a checkbox field](https://user-images.githubusercontent.com/2313998/164065184-e3c615a3-fb68-4878-b7f8-9a57efb748cb.png )
+
+### Label (required)
```

Comment:
In PRC, I see a prop called `layout` which I think is intended to allow the label to display on top of the input, or on the side (left) before the input. Am I correct in that assumption? The code looks a little broken right now:

![image](https://user-images.githubusercontent.com/18661030/167962742-ce31d59a-4246-4f6e-a118-be6a4dec2465.png)

If this is meant to be inline on the left, is that a pattern we need to support? If yes, we might want to document when you would use an inline label.

Reply:
This prop is meant to be used when you're using a custom input component, and you want the input to be laid out like a checkbox or radio form control. You can see an example in the docs: https://primer.style/react/FormControl#with-a-custom-input

I was stumped by this prop in the component API, as were the other PRC folks I consulted. Do you have any ideas @langermank ? At the very least, we shouldn't show this prop in Storybook stories.

Reply:
Got it, I totally thought this was implying a label inline vs. block.. so its more like a private API. I might have called the prop `labelPosition`? But I haven't thought it through too much 😄 in the meantime, you could always add a description to that prop in Storybook to explain what the does or maybe when to use it.